core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java

```diff
@@ -306,26 +307,23 @@ void build(List<StaticBytecodeRecorderBuildItem> staticInitTasks,
         CatchBlockCreator preventFurtherStepsBlock = tryBlock.addCatch(PreventFurtherStepsException.class);
         preventFurtherStepsBlock.invokeVirtualMethod(ofMethod(StartupContext.class, "close", void.class), startupContext);
 
+        CatchBlockCreator catchConfigurationException = tryBlock.addCatch(ConfigurationException.class);
+
+        // Handle Configuration Exception thrown at startup
+        handleThrownException(launchMode, catchConfigurationException);
+
+        catchConfigurationException.invokeStaticMethod(ofMethod(ApplicationStateNotification.class,
+                "notifyStartupFailed", void.class, Throwable.class),
+                catchConfigurationException.getCaughtException());
+        catchConfigurationException.invokeVirtualMethod(ofMethod(StartupContext.class,
+                "close", void.class), startupContext);
+        catchConfigurationException.throwException(catchConfigurationException.getCaughtException());
+        mv.returnValue(null);
```

Comment:
I think the return is already added further down, so this line should be removed. Otherwise the generated code will look like this:

```java
try {
...
}
catch (ConfigurationException e) {
...
}
catch (Throwable e) {
...
}
return null;
return null;
```